product_supplierinfo_for_customer_sale_quick/models/sale_order_line.py

```diff
@@ -0,0 +1,27 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    @api.onchange("name", "product_id")
+    def _onchange_product_id(self):
+        for line in self:
+            if line.product_id:
+                supplierinfo = line.product_id._select_customerinfo(
+                    partner=line.order_partner_id
+                )
+                if (
+                    supplierinfo
+                    and supplierinfo.product_code
+                    and supplierinfo.product_name
+                ):
+                    line.name = (
+                        "["
+                        + str(supplierinfo.product_code)
+                        + "]"
+                        + " "
+                        + str(supplierinfo.product_name)
+                    )
```

Comment:
**praise**: The `onchange` is an ok place for this, as `._complete_quick_line_vals()` calls every onchange linked to the fields. Good choice! 

**issue**: However, I don't think a change to `name` should trigger the onchange. I understand that just adding `product_id` to the `onchange` decorator doesn't trigger when you want it to with just `product_id`, but at worst maybe we could call the `onchange` manually. Besides, `product_supplierinfo_for_customer_sale` already implements this logic in `_compute_product_customer_code()`, so we shouldn't add the same logic again here.